src/v/datalake/translation/scheduling.cc

```diff
@@ -0,0 +1,443 @@
+        translation_duration
+          = std::chrono::duration_cast<std::chrono::milliseconds>(
+            clock::now() - state._current_translation_start_time.value());
+    }
+    fmt::print(
+      os,
+      "{{translator: {}, ms_since_start: {}, stop_in_progress: {}, running: "
+      "{}, waiting: {}}}",
+      *state._translator,
+      translation_duration,
+      state._stop_in_progress,
+      state._running_hook.is_linked(),
+      state._waiting_hook.is_linked());
+    return os;
+}
+
+void scheduler_state::start_translation(
+  translator_state& state, clock::duration time_slice) {
+    auto holder = gate.hold();
+    vassert(
+      state._waiting_hook.is_linked(),
+      "Invalid request to start translation. Translator not in waiting state: "
+      "{}",
+      state);
+    vassert(
+      !state._running_hook.is_linked(),
+      "Invalid request to start translation, Translator already in running "
+      "state: {}",
+      state);
+    try {
+        state._translator->start_translation(time_slice);
+    } catch (...) {
+        vlog(
+          datalake_log.warn,
+          "Exception {} starting translation: {}",
+          std::current_exception(),
+          state);
+        // Move it to the back of the waiting queue, to be retried later.
+        state._waiting_hook.unlink();
+        waiting.push_back(state);
+        return;
+    }
+    state._current_translation_start_time = clock::now();
+    state._waiting_hook.unlink();
+    running.push_back(state);
+}
+
+void scheduler_state::stop_translation(translator_state& state) {
+    auto holder = gate.hold();
+    vassert(
+      !state._waiting_hook.is_linked(),
+      "Invalid request to stop translation. Translator in waiting state: {}",
+      state);
+    vassert(
+      state._running_hook.is_linked(),
+      "Invalid request to stop translation. Translator not in running state {}",
+      state);
+    vassert(
+      !state._stop_in_progress,
+      "Duplicate stop request on translator: {}",
+      state);
+    vlog(datalake_log.debug, "stopping translator: {}", state);
+    try {
+        state._translator->stop_translation();
+        state._stop_in_progress = true;
+    } catch (...) {
+        vlog(
+          datalake_log.warn,
+          "Exception {} stopping translation: {}",
+          std::current_exception(),
+          state);
+    }
+}
+
+scheduler::scheduler(
+  size_t total_memory,
+  size_t memory_block_size,
+  std::unique_ptr<scheduling_policy> policy)
+  : _scheduling_policy(std::move(policy))
+  , _mem_tracker(reservations_tracker::make_default(
+      total_memory, memory_block_size, *this)) {
+    ssx::repeat_until_gate_closed_or_aborted(_state.gate, _state.as, [this] {
+        return main().handle_exception([](const std::exception_ptr& e) {
+            if (!ssx::is_shutdown_exception(e)) {
+                vlog(
+                  datalake_log.warn,
+                  "Encountered exception in main loop: {}",
+                  e);
```

Comment:
nit: maybe still log at debug when shutting down? Might save some debugging cycles, given seems like it'll be completely silent otherwise